src/Essential.Diagnostics.RollingFileTraceListener/Diagnostics/RollingFileTraceListener.cs

```diff
@@ -218,6 +260,7 @@ protected override void Write(string category, string message, object data)
             }
             else
             {
+                rollingTextWriter.FileSystem.CreateSubdirectories = CreateSubdirectories;
```

Comment:
Rather than set the value on every write/flush, it would be better to have lazy creation (instead of in the constructor) of the rolling text writer (with double checking locking for multi-threaded), and pass Create into the constructor. 

There are actually 2 PRs for adding features to the rolling writer, so setting it up with lazy creation will make it easier to add multiple configurable capabilities.

e.g.

```
protected RollingTextWriter RollingTextWriter { 
  get {
    if (rollingTextWriter == null)
    {
       lock (rollingTextWriterLock)
       {
          if (rollingTextWriter == null)
          {
             rollingTextWriter = new RollingTextWriter (... path ..., CreatFolders);
          }
        }
      }
    }
    return rollingTextWriter;
  }
}
```